packages/lightning-lsp-common/src/utils.ts

```diff
@@ -13,27 +13,68 @@ export const glob = promisify(Glob);
 
 const RESOURCES_DIR = 'resources';
 
+async function fileContainsLine(file: string, expectLine: string): Promise<boolean> {
+    const trimmed = expectLine.trim();
+    for (const line of (await fs.readFile(file, 'utf8')).split('\n')) {
+        if (line.trim() === trimmed) {
+            return true;
+        }
+    }
+    return false;
+}
+
+export function toResolvedPath(uri: string): string {
+    return resolve(URI.parse(uri).fsPath);
+}
+
+function isLWCRootDirectory(context: WorkspaceContext, uri: string): boolean {
+    if (context.type === WorkspaceType.SFDX) {
+        const file = toResolvedPath(uri);
+        return file.endsWith('lwc');
+    }
+    return false;
+}
+
+function isAuraDirectory(context: WorkspaceContext, uri: string): boolean {
+    if (context.type === WorkspaceType.SFDX) {
+        const file = toResolvedPath(uri);
+        return file.endsWith('aura');
+    }
+    return false;
+}
+
+export async function isLWCWatchedDirectory(context: WorkspaceContext, uri: string): Promise<boolean> {
+    const file = toResolvedPath(uri);
+    return await context.isFileInsideModulesRoots(file);
+}
+
+export async function isAuraWatchedDirectory(context: WorkspaceContext, uri: string): Promise<boolean> {
+    const file = toResolvedPath(uri);
+    return await context.isFileInsideAuraRoots(file);
+}
+
 /**
  * @return true if changes include a directory delete
  */
+// TODO This is not waiting for the response of the promise isLWCWatchedDirectory, maybe we have the same problem on includesDeletedAuraWatchedDirectory
 export function includesDeletedLwcWatchedDirectory(context: WorkspaceContext, changes: FileEvent[]): boolean {
     for (const event of changes) {
-        if (event.type === FileChangeType.Deleted && isLWCWatchedDirectory(context, event.uri)) {
+        if (event.type === FileChangeType.Deleted && event.uri.indexOf('.') === -1 && isLWCWatchedDirectory(context, event.uri)) {
```

Comment:
This should await on `isLWCWatchedDirectory` and will have to add `async` to the method signature.